content/en/docs/ops/integrations/opa/index.md

```diff
@@ -0,0 +1,178 @@
+---
+title: OPA
+description: Information on how to integrate with Open Policy Agent (OPA).
+weight: 20
+keywords: [integration,opa,authorization]
+aliases:
+  - /docs/tasks/traffic-management/ingress/opa/
+  - /docs/examples/advanced-gateways/opa/
+owner: istio/wg-environments-maintainers
+test: no
+---
+
+[Open Policy Agent (OPA)](https://www.openpolicyagent.org/) is a general-purpose policy engine that can be used to enforce policies across the stack. OPA can be integrated with Istio to enforce authorization policies.
+
+## Understanding Attribute-Based Access Control
+
+ABAC (Attribute-Based Access Control) is a model that defines access control based on attributes associated with the request, the resource, and the user. OPA can be used to enforce ABAC policies in Istio.
+
+Istio provides a way to enforce natively RBAC (Role-Based Access Control) policies. However, ABAC policies can be more flexible and can be used to enforce more complex policies.
+
+## How Istio and OPA Work Together
+
+Istio (istio-proxy) is based on [Envoy Proxy](https://www.envoyproxy.io/). certain capabilities of Envoy are configured in filters, being one of them the `ext_authz` filter. This filter can be used to call an external authorization service, like OPA.
+
+In reality, whichever service that implements the [Envoy's gRPC Authorization service](https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/auth/v3/external_auth.proto) can be used.
+
+{{< image width="75%"
+    link="./filters.png"
+    alt="Filters"
+    >}}
+
+OPA is one of the services that implement this gRPC service.
+
+When a request is made to an Istio service that is defined with an AuthorizationPolicy, the `ext_authz` filter is called. This filter will call OPA to check if the request is allowed. OPA will evaluate the request against the policies defined and return a decision to Envoy (istio-proxy).
+
+{{< image width="75%"
+    link="./opa1.png"
+    alt="OPA"
+    >}}
+
+### Advanced use-cases
+
+The Envoy's Authorization Service allows receiving from the Authorization server additional attributes that can be used not only to make the decision but also to enrich the request and the response.
+
+{{< text json >}}
+{
+  "headers": [],
+  "headers_to_remove": [],
+  "dynamic_metadata": {...},
+  "response_headers_to_add": [],
+  "query_parameters_to_set": [],
+  "query_parameters_to_remove": []
+}
+{{< /text >}}
+
+When OPA returns its response with these attributes, Envoy (istio-proxy) can modify the request and the response based on the returned attributes:
+
+- It can add headers to the request
+- It can remove headers from the request
+- It can add headers to the response
+- It can add query parameters to the request
+- It can remove query parameters from the request
+- It can add dynamic metadata to the request
```

Comment:
suggestion: I'd try and reduce the repetition of the 'it can' here.


```
When OPA returns its response with these attributes, Envoy (istio-proxy) can modify the request and the response based on the returned attributes by performing the following actions:

- Adding headers to the request
- Removing headers from the request
- Adding headers to the response
- Adding query parameters to the request
- Removing query parameters from the request
- Adding dynamic metadata to the request
```